### ISSUE TITLE
Combining ExecutionPlan and ExecutionEvent in KafkaLineageDispatcher

### ISSUE THREAD

**Issue by zacayd** [AbsaOSS/spline-spark-agent#540]
### Discussed in https://github.com/AbsaOSS/spline-spark-agent/discussions/446

<div type='discussions-op-text'>

<sup>Originally posted by **rkrumins** June 13, 2022</sup>
I am working on building out a service that takes the great work done by Spline Spark Agent in terms of capturing the lineage and metrics and eventually publishing that into the target metadata store. Spline Spark Agent works well so far. 

The idea here is to use the existing KafkaLineageDispatcher but given that messages are sent separately (ExecutionPlan as one followed by ExecutionEvent one), is there a specific reason why messages are not sent together? As in, in a single Kafka message that contains the "plan" and "event" parts? 

Therefore, before starting the implementation of a version of KafkaLineageDispatcher that combines messages from ExecutionPlan and ExecutionEvent or working on a custom utility that effectively joins both messages from a topic of my choice, could somebody please clarify why messages are sent separately? Also, is this done as per a specific set of edge cases? 

In terms of the existing codebase, I am referring to the following:
- https://github.com/AbsaOSS/spline-spark-agent/blob/develop/core/src/main/scala/za/co/absa/spline/harvester/dispatcher/KafkaLineageDispatcher.scala#L68
- https://github.com/AbsaOSS/spline-spark-agent/blob/develop/core/src/main/scala/za/co/absa/spline/harvester/dispatcher/KafkaLineageDispatcher.scala#L74</div>

**Comment by wajda** [comment]
@zacayd, I understand that you might have a lot of questions around Spline as well as other related technologies. But please, try to be more organized in what, how and where you post on GitHub. I noticed some of your messages are off-topics, or even totally confusing and meaningless like this one. Please save our and other community members' time by posting more clear questions into related issues/discussions. 

P.S. and please, stop emailing me directly with your questions. Me and the rest of the team do read GitHub.

**Comment by zacayd** [comment]
@wajda  i am new to the spline system - and  my way to learn is by asking questions
and i see i am getting lost - so can you support or not?

**Comment by wajda** [comment]
It's OK to ask as many questions as needed and we are happy to answer them. All I'm asking is to be more organized and clear. What problem did you want to address in this current ticket? You copy/pasted another person question from an unrelated discussion with no additional comment from your side? What did you expect us to do with this?

**Comment by zacayd** [comment]
my question realtes to using the kafka with databricks - i sent a detailed question of the issue- becaue it relared to kafka in spline

**Comment by wajda** [comment]
I have answered to your question in another issue you created - AbsaOSS/spline-getting-started#38
